napari/layers/points/_points_mouse_bindings.py

```diff
@@ -139,3 +138,83 @@ def _toggle_selected(selected_data, value):
         selected_data.add(value)
 
     return selected_data
+
+
+def _update_drag_vectors_from_event(layer, event):
+    """Update the drag normal and drag up vectors on the Points layer from
+    a mouse event object.
+
+    Note that in 2D mode, the layer._drag_normal and layer._drag_up
+    are set to None
+
+    Parameters
+    ----------
+    layer : "napari.layers.Points"
+        The Points layer to update
+    event
+        The mouse event object
+    """
+    n_display = len(event.dims_displayed)
+    if n_display == 3:
+        ndim_world = len(event.position)
+        # if in 3D, set the drag normal and up directions
+        # get the indices of the displayed dimensions
+        dims_displayed_data = layer._world_to_data_dims_displayed(
+            dims_displayed=event.dims_displayed, ndim_world=ndim_world
+        )
+
+        # get the view direction in data coordinates
+        layer._drag_normal = np.asarray(
+            layer._world_to_data_ray(event.view_direction)
+        )[dims_displayed_data]
+
+        # get the up direction of the camera
+        layer._drag_up = np.asarray(
+            layer._world_to_data_ray(event.up_direction)
+        )[dims_displayed_data]
```

Comment:
minor nitpick, maybe packing this into a util function would increase readability?